File: i2f-jdk/i2f-swl/src/main/java/i2f/swl/std/ISwlAsymmetricEncryptor.java
package i2f.swl.std;

import i2f.jce.std.encrypt.asymmetric.key.AsymKeyPair;

/**
 * @author Ice2Faith
 * @date 2024/7/9 18:41
 * @desc
 */
public interface ISwlAsymmetricEncryptor {
    AsymKeyPair generateKeyPair();

    AsymKeyPair getKeyPair();

    void setKeyPair(AsymKeyPair key);

    String getPublicKey();

    void setPublicKey(String publicKey);

    String getPrivateKey();

    void setPrivateKey(String privateKey);

    String encrypt(String data);

    String decrypt(String data);

    String sign(String data);

    boolean verify(String sign, String data);
}

File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-swl/core/std/ISwlAsymmetricEncryptor.js
/**
 * @return {ISwlAsymmetricEncryptor}
 * @interface ISwlAsymmetricEncryptor
 */
function ISwlAsymmetricEncryptor() {
    return {
        /**
         * @return {AsymKeyPair}
         */
        generateKeyPair() {

        },
        /**
         * @return {AsymKeyPair}
         */
        getKeyPair() {

        },
        /**
         * @param key {AsymKeyPair}
         * @return {void}
         */
        setKeyPair(key) {

        },
        /**
         * @return {String}
         */
        getPublicKey() {

        },
        /**
         * @param publicKey {String}
         * @return {void}
         */
        setPublicKey(publicKey) {

        },
        /**
         * @return {String}
         */
        getPrivateKey() {

        },
        /**
         *
         * @param privateKey {String}
         * @return {void}
         */
        setPrivateKey(privateKey) {

        },
        /**
         *
         * @param data {String}
         * @return {String}
         */
        encrypt(data) {

        },
        /**
         *
         * @param data {String}
         * @return {String}
         */
        decrypt(data) {

        },
        /**
         *
         * @param data {String}
         * @return {String}
         */
        sign(data) {

        },
        /**
         *
         * @param sign {String}
         * @param data {String}
         * @return {boolean}
         */
        verify(sign, data) {

        }
    }
}

export default ISwlAsymmetricEncryptor
